packages/tjdashboard/administrator/models/forms/widget.xml

```diff
@@ -8,25 +8,57 @@
 			description="JGLOBAL_FIELD_ID_DESC"/>
 
 		<field
-			name="dashboard_id" 
+			name="dashboard_id"
 			type="sql"
 			label="COM_TJDASHBOARD_WIDGET_FORM_LBL_DASHBOARDID"
 			description="COM_TJDASHBOARD_WIDGET_FORM_DESC_DASHBOARDID"
 			class="inputbox"
 			query="SELECT dashboard_id as value, title as dashboard_id FROM #__tj_dashboards WHERE `state`=1 ORDER BY `ordering`"
 			required="true" />
 
+
+
 		<field
 			name="title"
-			filter="string" 
+			filter="string"
 			type="text"
 			label="COM_TJDASHBOARD_WIDGET_FORM_LBL_TITLE"
 			description="COM_TJDASHBOARD_WIDGET_FORM_DESC_TITLE"
 			size="10"
 			required="true" />
 
 		<field
-			name="data_plugin" 
+			name="primary_text"
+			filter="string"
+			type="text"
+			label="COM_TJDASHBOARD_WIDGET_FORM_LBL_PRIMARY_TEXT"
+			description="COM_TJDASHBOARD_WIDGET_FORM_DESC_PRIMARY_TEXT"
+			size="15"/>
+
+		<field
```

Comment:
@monikaatvowel This should not the property of the widget. The UI is the property of renderer.

cc
@coolbung

Reply:
@thite-amol fixed now. the code is moved to renderer side and now using params field